caddyhttp/basicauth/basicauth_test.go

```diff
@@ -14,19 +14,31 @@ import (
 )
 
 func TestBasicAuth(t *testing.T) {
+	var i int
+	// This handler is registered for tests in which the only authorized user is
+	// "okuser"
+	upstreamHandler := func(w http.ResponseWriter, r *http.Request) (int, error) {
+		remoteUser, _ := r.Context().Value("remote_user").(string)
```

Comment:
The `, _` isn't necessary for type assertions, but why ignore the bool/error? Wouldn't it panic then? I guess it's a test case so it's not the end of the world, but perhaps better to check the error and t.Fatal if there's a problem.

Edit: Nevermind, it doesn't panic. Clever.